plugin/src/main/java/io/cucumber/plugin/event/TestRunFinished.java

```diff
@@ -7,7 +7,18 @@
 @API(status = API.Status.STABLE)
 public final class TestRunFinished extends TimeStampedEvent {
 
+    private final Exception exception;
```

Comment:
Wrap this in a result object. Add state.

Consider doing the same for the messages.